src/libraries/System.Net.Quic/ref/System.Net.Quic.cs

```diff
@@ -128,6 +128,13 @@ public partial class QuicStreamAbortedException : System.Net.Quic.QuicException
         public QuicStreamAbortedException(string message, long errorCode) : base (default(string)) { }
         public long ErrorCode { get { throw null; } }
     }
+
+    public partial class QuicTlsVersionException : System.Net.Quic.QuicException
```

Comment:
Should we add a new exception type for this? Also this is basically a new API surface. I'd prefer to use the existing exception type. 
What's the motivation behind introducing it?

Reply:
It seemed cleaner to me than the equivalent using the existing type which would be:
```c#
throw new QuicException(SR.net_ssl_app_protocols_invalid, null, (int)status);
```

And follows the existing pattern for Exceptions that use strings from resources.

I can appreciate not wanting to expand the API surface just for this, I'll add a helper to the `QuicExceptionHelpers` for the above and remove the public exception.